packages/path_provider/path_provider_android/test/path_provider_android_test.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:path_provider_android/path_provider_android.dart';
+import 'package:path_provider_platform_interface/src/enums.dart';
```

Comment:
Not too sure about importing path_provider_platform_interface/src/enums.dart to reference StorageDirectory. Since this enum is only applicable to Android, should I move it into the path_provider_android package?

Reply:
> Not too sure about importing path_provider_platform_interface/src/enums.dart to reference StorageDirectory.

It's re-exported from the main package header (`package:path_provider_platform_interface.dart`), which is what you should include instead.

> Since this enum is only applicable to Android, should I move it into the path_provider_android package?

You can't, because the platform interface uses it. The API can't use a type that's not defined anywhere.

If path_provider were being written from scratch as a federated plugin, the Android-specific API would be an extension that only exited on the Android implementation, in which case the enum could move, but moving it now would be a breaking change that's not worth the effect that would have on the ecosystem.

Reply:
Got it. I've changed it to import from the main package header. Any other issues?